implementation.js

```diff
@@ -78,7 +78,7 @@ if (hasSymbols) {
 		iteratorSymbol = '@@iterator';
 	} else {
 		try {
-			if (Function('for (var x of [0]) x;')() === 0) { // eslint-disable-line no-new-func
+			if (Function('for (var x of new Set().add(0)) return x;')() === 0) { // eslint-disable-line no-new-func
 				forOf = Function('iterable', 'var arr = []; for (var value of iterable) arr.push(value); return arr;'); // eslint-disable-line no-new-func
```

Comment:
i think this might work better by defining `forOf` locally, and then *calling* that function with `const s = new Set(); s.add(0);` (ie, not relying on `.add` returning `this`, since not every implementation does) and then we'd only need one `Function`.